### PR TITLE
Align ARIAMixin with changes in HTML

### PR DESCRIPTION
**Pull request by annevk**
This does not address most of the issues described in #1110, but does provide a better base for ElementInternals support and tackling the issues in #1110.

Corresponding HTML change: https://github.com/whatwg/html/pull/8496.

---

This is meant to be editorial when combined with the HTML PR landing around the same time, leaving existing spec-UB as spec-UB. Please do not merge this before the HTML PR is merged.

cc @domenic

***
<a href="https://pr-preview.s3.amazonaws.com/annevk/aria/pull/1876.html" title="Last updated on Feb 15, 2024, 7:46 PM UTC (7718379)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/aria/1876/9bf6e21...annevk:7718379.html" title="Last updated on Feb 15, 2024, 7:46 PM UTC (7718379)">Diff</a>

### CODE REVIEW

**Comment by pkra** [comment]
Converting to draft until upstream PR is resolved.

**Review by domenic — APPROVED** [review]
LGTM, I guess this did dramatically simplify things.

**Comment by annevk** [comment]
I think it will have to get more complicated as we address the remaining issues around `DOMString?` and such, but for now it's indeed a simplification.

**Comment by domenic** [comment]
Can anyone work on merging this? I got very confused implementing the current spec in jsdom.

File: index.html [hunk 1]
```diff
@@ -13680,32 +13680,18 @@ <h2>Interface Mixin <dfn>ARIAMixin</dfn></h2>
...
-
-		<p>For every IDL attribute <var>idlAttribute</var> defined in <code>ARIAMixin</code>, on getting, it must perform the following steps:</p>
+		<p>For every IDL attribute <var>idlAttribute</var> defined in <code>ARIAMixin</code> when included on an element:</p>
 
 		<ol>
-			<li><p>Let <var>contentAttribute</var> be the ARIA content attribute determined by looking up <var>idlAttribute</var> in the ARIA Attribute Correspondence table.</p></li>
+			<li><p>Let <var>contentAttributeName</var> be the local name of the ARIA content attribute determined by looking up <var>idlAttribute</var> in the ARIA Attribute Correspondence table.</p></li>
 
-			<li><p>Return the result of running the <a><code>ARIAMixin</code> getter steps</a>, given this, <var>idlAttribute</var>, and <var>contentAttribute</var>.</p></li>
+			<li><p><var>idlAttribute</var> must <a>reflect</a> <var>contentAttributeName</var> and <a>support <code>ElementInternals</code></a>.</p></li>
```
Comment (keithamus) [inline index.html:13691]: Looks like `reflect` and `support ElementInternals` references are missing.

**Review by keithamus — APPROVED** [review]
This looks like a straightforward and worthwhile change to me, once the above comment is resolved.

File: index.html [hunk 2]
```diff
@@ -13680,32 +13680,18 @@ <h2>Interface Mixin <dfn>ARIAMixin</dfn></h2>
...
-		<p>Similarly, on setting, it must perform the following steps:</p>
-
-		<ol>
-			<li><p>Let <var>contentAttribute</var> be the ARIA content attribute determined by looking up <var>idlAttribute</var> in the ARIA Attribute Correspondence table.</p></li>
-
-			<li><p>Run the <a><code>ARIAMixin</code> setter steps</a>, given this, <var>idlAttribute</var>, <var>contentAttribute</var>, and the given value.</p></li>
-		</ol>
-
-		<p class="note">This very general framework is motivated by the desire for different host interfaces, such as <code>Element</code> and <code>ElementInternals</code>, to give these IDL attributes different behaviors. The alternative is requiring each host interface to duplicate the IDL attributes independently, so that they can specify independent behaviors, but that comes with a high risk of them getting out of sync.</p>
+		<p class="note">In practice, this means that, e.g., the <code>role</code> IDL attribute on <code>Element</code> reflects the <code>role</code> content attribute; the <code>ariaValueMin</code> IDL attribute reflects the <code>aria-valuemin</code> content attribute; etc.</p>
```
Comment (cookiecrook) [inline index.html:13694]:
```suggestion
		<p class="note">In practice, this means that, e.g., the <code>role</code> IDL attribute on <code>Element</code> reflects the <code>role</code> content attribute; the <code>ariaValueMin</code> IDL attribute reflects the <pref>aria-valuemin</pref> content attribute; etc. Ambiguity clarifications (such as <code>ariaPosInSet</code>) are listed in <a href="https://w3c.github.io/aria/#idl_attr_exceptions">IDL Attribute Name Notes or Exceptions</a>.</p>
```

File: index.html (hunk 1 above)
Comment (cookiecrook) [reply index.html:13691]:
```suggestion
			<li><p><var>idlAttribute</var> must <a href="https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#reflect">reflect</a> <var>contentAttributeName</var> and <a href="https://html.spec.whatwg.org/multipage/custom-elements.html#the-elementinternals-interface">support <code>ElementInternals</code></a>.</p></li>
```

**Review by cookiecrook — APPROVED** [review]
Approve with included edits.

File: index.html (hunk 1 above)
Comment (cookiecrook) [reply index.html:13691]: @pkra move these refs to the common files if needed.

Comment (pkra) [reply index.html:13691]: @cookiecrook @keithamus this is just a regular exported term, see https://github.com/w3c/aria#references-to-other-w3c-specs for guidance but in short `[=reflect=]` and `{{ElementInternals}}` should work (there are other references in this style throughout the spec, too).

File: index.html (hunk 2 above)
Comment (spectranaut) [reply index.html:13694]: Good edit, imo, but use a relative path to idl_attr_exceptions:
```suggestion
		<p class="note">In practice, this means that, e.g., the <code>role</code> IDL attribute on <code>Element</code> reflects the <code>role</code> content attribute; the <code>ariaValueMin</code> IDL attribute reflects the <pref>aria-valuemin</pref> content attribute; etc. Ambiguity clarifications (such as <code>ariaPosInSet</code>) are listed in <a href="#idl_attr_exceptions">IDL Attribute Name Notes or Exceptions</a>.</p>
```

File: index.html (hunk 1 above)
Comment (spectranaut) [reply index.html:13691]:
```suggestion
			<li><p><var>idlAttribute</var> must [=reflect=] <var>contentAttributeName</var> and support {{ElementInternals}}.</p></li>
```

**Review by spectranaut — COMMENTED** [review]
Applied edits on @cookiecrook's edits.

**Pull request merged**